prusti-specs/src/lib.rs

```diff
@@ -118,12 +119,22 @@ fn generate_for_requires(attr: TokenStream, item: &untyped::AnyFnItem) -> Genera
     ))
 }
 
-/// Generate spec items and attributes to typecheck th and later retrieve "ensures" annotations.
+/// Generate spec items and attributes to typecheck and later retrieve "ensures" annotations.
 fn generate_for_ensures(attr: TokenStream, item: &untyped::AnyFnItem) -> GeneratedResult {
+    common_ensures(attr, item)
+}
+
+/// Generate spec items and attributes to typecheck and later retrieve "ensures_result_eq" annotations.
+fn generate_for_ensures_result_eq(attr: TokenStream, item: &untyped::AnyFnItem) -> GeneratedResult {
+    let assertion = parse_quote_spanned!{item.span() => result == (#attr)};
```

Comment:
This is not enough, as it still produces e.g. `result == forall(...)` and expects the preparser to be able to deal with it, which it cannot. The approach here is to parse the attribute as an *expression*, then add the `result == ...` over the parsed tree (without relying on the preparser).

Reply:
and by *expression*, you mean a `prusti_specs::specifications:untyped::Assertion`, right?

Reply:
Yeah, sorry, I was confused due to this being different in VIR. But this raises a good point, I'm not sure you can actually build an assertion that is `AssertionKind::Expr` representing `result == <another assertion>`. I'll need to think about this a bit more…

Reply:
put up a WIP of reworking to handle predicates separately all the way through, still some work needed.